openedx/core/djangoapps/user_authn/views/register.py

```diff
@@ -276,7 +285,6 @@ def create_account_with_params(request, params):  # pylint: disable=too-many-sta
 
     try:
         _record_registration_attributions(request, new_user)
-        _record_is_marketable_attribute(is_marketable, new_user)
     # Don't prevent a user from registering due to attribution errors.
     except Exception:   # pylint: disable=broad-except
         log.exception('Error while attributing cookies to user registration.')
```

Comment:
Can we move this complete block above instead of just marketable attribute creation? I believe two separate try/excepts are unnecessary.

Reply:
This exception is for cookies. 
My only concern was that if an error occurs in `_record_registration_attributions` then `_record_is_marketable_attribute` won't even run.

Reply:
Makes sense 👍